Надо подкачаться

lisenkovnikita opened the pull request:
new file:   js/api.js
modified:   js/change-photo.js
deleted:    js/data.js
modified:   js/form.js
modified:   js/main.js
modified:   js/rendering.js
modified:   js/util.js

---
:mortar_board: [Надо подкачаться](https://up.htmlacademy.ru/javascript/27/user/2054309/tasks/18)

:boom: https://htmlacademy-javascript.github.io/2054309-kekstagram-simple-27/9/

lisenkovnikita committed:
11.16. Надо подкачаться
new file:   js/api.js
modified:   js/change-photo.js
deleted:    js/data.js
modified:   js/form.js
modified:   js/main.js
modified:   js/rendering.js
modified:   js/util.js

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,46 @@
+import {userPhotosLoadError} from './util.js';
+import {showOkUpload, showFailUpload, enableUploadButton} from './form.js';
+
+const getData = () => {
```
если это API, то оно ничего не знает про рендер. Можно передать сюда обработчики onSuccess и onError в аргументах

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,46 @@
+    });
+};
+
+const sendData = (onSuccess, evt) => {
```
аналогично, onError аргументом, и событие evt тут не нужно, сразу event.target передавай

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,46 @@
+  )
+    .then((response) => {
+      if (response.ok) {
+        onSuccess();
+        showOkUpload();
```
showOk должен выполняться внутри onSuccess

frontstall commented on js/util.js:
```diff
@@ -1,16 +1,26 @@
+const isEscapeKey = (evt) => evt.key === 'Escape';
 
-const getRandomArrayElement = (elements) => elements[getRandomPositiveInteger(0, elements.length - 1)];
+const userPhotosLoadError = () => {
```
функции начинаются с глагола

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,46 @@
+        userPhotosLoadError();
+      }
+    })
+    .then((photos) => {
```
просто .then(renderPhotos) и .catch(userPhotosLoadError)

lisenkovnikita committed:
11.16. Надо подкачаться
modified:   js/api.js
modified:   js/form.js
modified:   js/main.js
modified:   js/util.js

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,39 @@
+    .catch(onError);
+};
+
+const sendData = (onSuccess, onError, eventTarget) => {
```
апи ничего не должно знать о том, что за данные оно отправляет. Просто data (уже в FormData), параметры запроса выглядят так:

```js
{
  method: 'POST',
  body: data,
}
```

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,39 @@
+
+const sendData = (onSuccess, onError, eventTarget) => {
+  fetch(
+    'https://27.javascript.pages.academy/kekstagram-simple',
```
адрес в константу, все адреса должны быть там. Удобно сделать это в виде объекта.

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,39 @@
+        onError();
+      }
+    })
+    .catch(() => {
```
`catch(onError);`

frontstall commented on js/api.js:
```diff
@@ -0,0 +1,39 @@
+    .catch(() => {
+      onError();
+    })
+    .finally(() => {
```
`finally(enableUploadButton)`

frontstall commented on js/form.js:
```diff
@@ -1,19 +1,66 @@
 
+const showOkUpload = () => {
+  const successElement = uploadSuccessTemplate.cloneNode(true);
+  bodyElement.appendChild(successElement);
```
сначала наполняем элемент, потом добавляем на страницу. Иначе каждое изменение приводит к перерисовке страницы

frontstall commented on js/form.js:
```diff
@@ -50,12 +97,65 @@ const validateData = new Pristine(buttonForPost, {
-buttonForPost.addEventListener('submit', (evt) => {
+const showFailUpload = () => {
+  const errorElement = uploadErrorTemplate.cloneNode(true);
+  bodyElement.appendChild(errorElement);
```
same
